internal/removals/remove.go

```diff
@@ -164,6 +174,28 @@ func GetRemainingMap(logger logr.Logger, adminClient fdbadminclient.AdminClient,
 	return remainingMap, nil
 }
 
+// verifyExcludedProcesses will verify that the processes that should be excluded based on the FoundationDB machine-readble status are also excluded when issuing
+// the exclude command.
+func verifyExcludedProcesses(adminClient fdbadminclient.AdminClient, remaining []fdbv1beta2.ProcessAddress, addresses []fdbv1beta2.ProcessAddress) error {
+	remainingMap := map[string]fdbv1beta2.None{}
+
+	for _, remainingAddress := range remaining {
+		remainingMap[remainingAddress.MachineAddress()] = fdbv1beta2.None{}
+	}
+
+	checkExclusions := make([]fdbv1beta2.ProcessAddress, 0, len(addresses))
+	for _, addr := range addresses {
+		_, ok := remainingMap[addr.MachineAddress()]
+		if ok {
+			continue
+		}
+
+		checkExclusions = append(checkExclusions, addr)
+	}
+
+	return adminClient.ExcludeProcesses(checkExclusions)
```

Comment:
This will force a cluster recovery?

Reply:
Only if the cluster is not fully recovered: https://github.com/apple/foundationdb/blame/03b102d86aecbe700aa8402ae31d0431bfb0b2b9/fdbserver/ClusterRecovery.actor.cpp#L589